packages/wrangler/src/api/integrations/bindings/services.ts

```diff
@@ -0,0 +1,108 @@
+import { Response } from "miniflare";
+import { fetch, type Request } from "undici";
+import { getRegisteredWorkers } from "../../../dev-registry";
+import type { WorkerDefinition, WorkerRegistry } from "../../../dev-registry";
+import type { RequestInit } from "undici";
+
+/**
+ * Gets service bindings that proxy requests to external workers running locally, so that they
+ * can be passed to miniflare and used to fetch from such workers by the next app.
+ *
+ * @param services the services requested by the user
+ * @returns the service bindings ready to be passed to miniflare
+ */
+export async function getServiceBindings(
+	services:
+		| {
+				binding: string;
+				service: string;
+		  }[]
+		| undefined = []
+): Promise<ServiceBindings | undefined> {
+	if (services.length === 0) {
+		return;
+	}
+
+	let registeredWorkers: WorkerRegistry | undefined;
+
+	try {
+		registeredWorkers = await getRegisteredWorkers();
+	} catch {
+		/* */
+	}
+
+	if (!registeredWorkers) {
+		return;
+	}
```

Comment:
If you move this into its own function then you can avoid the `no-non-null-assertion` issue below.

```suggestion
	const registeredWorkers = await maybGetRegisteredWorkers();
	if (!registeredWorkers) {
		return;
	}
```

And elsewhere in this file:
```ts
async function maybeGetRegisteredWorkers() {
	try {
		return await getRegisteredWorkers();
  } catch {
    return undefined;
  }
}
```

Reply:
Change applied, thanks! 😄